Automattic/jetpack
# Search package: move search dashboard to package ( the base ) (1/2)

#### Changes proposed in this Pull Request:
The PR is a combination of #21840, #21841 and #21842, which includes the dependencies duplicated from Jetpack and a new state store. The PR doesn't touch the Jetpack the plugin (except for a dependency update) and will serve as a base to migrate the search dashboard on.

As the individual PRs were all reviewed by the search team already ( and most the team is on AFK now ), so I'm sending this for crew review straightly.

#21840 ([commit](https://github.com/Automattic/jetpack/pull/21865/commits/a9f40e8d2a448a280005ae2bfce14389b55d7157))
- Copied dependencies from Jetpack the plugin with zero change as a base. tests and state are skipped as we'll add new.

#21841 ([commit](https://github.com/Automattic/jetpack/pull/21865/commits/a773e56ba7617ca97ef64b52c7acbfd0e81a36c3))
- Added new state store based on @wordpress/data to replace the state store in Jetpack client. 

#21842 ([commit](https://github.com/Automattic/jetpack/pull/21865/commits/c255b3e24c271c8dfd5c232c0f807e2f1ad040d1))
- Removed unnecessary dependencies from copied components.
- Added simple tests for components
- Make form toggle component grid layout compatible

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
- Run `(cd projects/packages/search && pnpm test)`
- Run `(cd projects/packages/search && composer phpunit)`
- Ensure all tests pass
- Ensure no changes to Jetpack the plugin except for a dependency version bump
